r25523: #16967: error: no template named 'unary_function' in namespace 'std' when building in C++17 mode
https://bugs.scribus.net/view.php?id=16967
--- scribus/desaxe/digester.cpp	(revision 25522)
+++ scribus/desaxe/digester.cpp	(revision 25523)
@@ -448,10 +448,10 @@
 	return nfa;
 }
 
-struct CreateDFAState : public std::unary_function <std::set<nfa_state_t>, dfa_state_t> {
-
+struct CreateDFAState
+{
 	CreateDFAState(const std::vector<rule_t>& rules, const std::vector<nfa_state_t>& accepting) 
-	: n(0), rules_(rules), accepting_(accepting) 
+	: rules_(rules), accepting_(accepting) 
 	{}
 
 	dfa_state_t operator()(const std::set<nfa_state_t>& combination)
@@ -465,7 +465,7 @@
 		return result;
 	} 
 
-	unsigned int n;
+	unsigned int n { 0 };
 	const std::vector<rule_t>& rules_;
 	const std::vector<nfa_state_t>& accepting_;
 };
